src/sync_codec.rs

```diff
@@ -128,3 +127,22 @@ pub trait Decoder {
         Framed::new(io, self)
     }
 }
+
+pub trait Encoder<Item> {
```

Comment:
I suggest we use `tokio_util::codec::Encoder` so that users' impl will work here.